### PR TITLE
feat(sink): Add privatelink support for Kafka sink

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
- Add PrivateLink support for the Kafka sink in the cloud environment.
When users want to access a Kafka or AWS MSK in a different VPC, they can create a privatelink connection first, then use  the connection to create a Kafka sink.
- Resolve the privatelink connection info in the frontend instead of in Meta

```sql
CREATE SINK sink2 FROM mv2
WITH (
   connector='kafka',
   type='append-only',
   properties.bootstrap.server='b-1.xxx.amazonaws.com:9092,b-2.test.xxx.amazonaws.com:9092',
   topic='msk_topic',
   force_append_only='true',
   connection.name = 'connection1',
   privatelink.targets = '[{"port": 8001}, {"port": 8002}]',
);

```

related: https://github.com/risingwavelabs/risingwave/pull/9119, https://github.com/risingwavelabs/risingwave/issues/8771

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
~- [ ] I have added necessary unit tests and integration tests~
~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.
- Connector (sources & sinks)
- SQL commands, functions, and operators

### Release note
Add PrivateLink support for the Kafka sink in the cloud environment.

</details>

### CODE REVIEW

File: src/frontend/src/handler/create_sink.rs [hunk 1]
```diff
@@ -92,14 +95,29 @@ pub fn gen_sink_plan(
...
+        {
+            let conn = session.get_connection_by_name(sink_schema_name, &connection_name)?;
+            resolve_private_link_connection(&conn, properties)?;
+            Some(ConnectionId(conn.id))
+        } else {
+            None
+        }
+    };
+
+    info!("Create sink with connection {:?}", connection_id);
```
Comment (yezizp2012) [inline src/frontend/src/handler/create_sink.rs:113]:
```suggestion
    debug!("Create sink with connection {:?}", connection_id);
```

**Review by yezizp2012 — COMMENTED** [review]
We need to validate connection existence in meta side like `create source` and maintain connection dependency information.

**Comment by StrikeW** [comment]
> We need to validate connection existence in meta side like `create source` and maintain connection dependency information.

I already validated the connection in the Meta side. Am I miss something?
https://github.com/risingwavelabs/risingwave/pull/9263/files#diff-265655b0ac6123740dc95321546433436492c0b7355644ba505c6d62dbb70e5fR215-R219

**Comment by yezizp2012** [comment]
> > We need to validate connection existence in meta side like `create source` and maintain connection dependency information.
> 
> I already validated the connection in the Meta side. Am I miss something? @yezizp2012 [#9263 (files)](https://github.com/risingwavelabs/risingwave/pull/9263/files#diff-265655b0ac6123740dc95321546433436492c0b7355644ba505c6d62dbb70e5fR215-R219)

Oh sorry, I wasn't clear. I mean in `CatalogManager` the validation and dependency maintaining are still lacked before committing to meta store. You can refer to the implementation of create source:
https://github.com/risingwavelabs/risingwave/blob/21bbc4776c773a25eb51a145d7f60d610b3afd24/src/meta/src/manager/catalog/mod.rs#L1296-L1303

https://github.com/risingwavelabs/risingwave/blob/21bbc4776c773a25eb51a145d7f60d610b3afd24/src/meta/src/manager/catalog/mod.rs#L1358-L1361

File: src/connector/src/source/kafka/private_link.rs [hunk 2]
```diff
@@ -17,15 +17,44 @@ use std::str::FromStr;
...
 
-impl PrivateLinkConsumerContext {
-    pub fn new(broker_rewrite_map: Option<HashMap<String, String>>) -> anyhow::Result<Self> {
+impl BrokerAddrRewriter {
+    fn rewrite_broker_addr(&self, addr: BrokerAddr) -> BrokerAddr {
+        let reworte_addr = match self.rewrite_map.get(&addr) {
+            None => addr,
+            Some(new_addr) => new_addr.clone(),
+        };
+        reworte_addr
```
Comment (yezizp2012) [inline src/connector/src/source/kafka/private_link.rs:50]:
```suggestion
        let rewrote_addr = match self.rewrite_map.get(&addr) {
            None => addr,
            Some(new_addr) => new_addr.clone(),
        };
        rewrote_addr
```

File: src/connector/src/source/kafka/private_link.rs [hunk 3]
```diff
@@ -17,15 +17,44 @@ use std::str::FromStr;
...
+    }
+
+    pub fn new(
+        role: PrivateLinkContextRole,
+        broker_rewrite_map: Option<HashMap<String, String>>,
+    ) -> anyhow::Result<Self> {
+        tracing::info!("[{}] rewrite map {:?}", role, broker_rewrite_map);
```
Comment (yezizp2012) [inline src/connector/src/source/kafka/private_link.rs:57]:
```suggestion
        tracing::debug!("[{}] rewrite map {:?}", role, broker_rewrite_map);
```

**Review by yezizp2012 — APPROVED** [review]
Generally LGTM except some NIT.

**Pull request merged**